viewing_party/party.py

```diff
@@ -1,23 +1,239 @@
 # ------------- WAVE 1 --------------------
-
 def create_movie(title, genre, rating):
-    pass
+    """
+    Create a movie dictionary with title, genre, and rating.
+    Return None if any field is missing.
+    """
+    if not title or not genre or not rating:
+        return None
+    movie = {"title": title,
+            "genre": genre,
+            "rating": rating}
+    return movie
+
+def add_to_watched(user_data, movie):
+    """
+    Add a movie to the user's watched list and return updated user_data.
+    """
+    user_data["watched"].append(movie)    
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    """
+    Add a movie to the user's watchlist and return updated user_data.
+    """
+    user_data["watchlist"].append(movie)
+    return user_data
 
+def watch_movie(user_data, title):
+    """Create a dictionary that represents a single movie. If any inputs are falsy, return None.
+    Args:
+        title (str): Title of movie.
+        genre (str): Gener of movie.
+        rating (float): Rating given to movie.
+
+    Returns:
+        dict: represents a movie with information about its title, genre, and rating.
+    """
+    list_movies = user_data["watchlist"]
+    for movie in list_movies[:]:
+        if movie["title"] == title:
+            list_movies.remove(movie)
+            user_data["watched"].append(movie)
+            break 
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    """
+    Return the average rating of all movies in the user's watched list.
+    If the list is empty, return 0.0.
+    """
+    rating_total = 0.0
+    list_movies = user_data["watched"]
+    if not list_movies:
+        return 0.0
+    for movie in list_movies:
+        rating_total += movie["rating"]
+    return rating_total / len(list_movies)    

```

Comment:
Consider adding a few blank lines within your function to separate logically distinct sections of the code (e.g., setup, processing, return).  According to [PEP 8](https://peps.python.org/pep-0008/#blank-lines), blank lines can improve readability by visually grouping related operations. Right now, everything runs together, which can make it harder to quickly understand the structure and flow of your function.
```suggestion
    rating_total = 0.0
    list_movies = user_data["watched"]
    
    if not list_movies:
        return 0.0
        
    for movie in list_movies:
        rating_total += movie["rating"]
        
    return rating_total / len(list_movies)    
```